infra/experimental/chronos/prepare-replay-rebuild

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/bash
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+PROJECT=$1
+{
+    echo "ENV REPLAY_ENABLED=1";
```

Comment:
We can just do this via the cloudbuild by passing `--change` to `docker commit` here: https://github.com/google/oss-fuzz/blob/d5619e1f7a87b7a131e867d6e0c1d0a3a2d3ebbd/infra/experimental/chronos/cloudbuild.yaml#L63

Reply:
I think we need to use this type of logic when dealing with docker images? `docker commit` is only for containers

Reply:
`commit` turns a container into an image.

This is probably easier overall than trying to pull out all the state ourselves and put it back in via `docker build`. Can we revert to doing that?

Reply:
> commit turns a container into an image.

But we don't have a container to work from at this point in the script -- the places where we use `prepare-ccache` it's to build an image from the `Dockerfile` in the project's dir -- we're not working from a container at these points

Reply:
we do have a container don't we?

```
- name: 'gcr.io/cloud-builders/docker'
  args:
  - run
  - --entrypoint=/bin/sh
  - --env=SANITIZER=address
  - --env=STORE_CCACHE=1
  - --env=COMPILE_USING_CCACHE=1
  - --name=${_PROJECT}-origin-asan
  - gcr.io/oss-fuzz/${_PROJECT}
  - -c
  - "compile && rm -rf /out/*"
  ```

From the above step, we have a container created with name `${_PROJECT}-origin-asan`. We can just commit that.

Reply:
but we want the container as it is prior to running `compile` for ccache builds?

Reply:
OK, got it. I got confused here because the code here is talking about replay script -- in that case we can just `commit` right?

Reply:
yeah, apologies -- the PR got a bit changed as we clarified the strategy.

The `cloudbuild.yaml` currently does the steps:

https://github.com/google/oss-fuzz/pull/12608/#discussion_r1830220783

Am not sure how we can shorten it though. We need to generate two caches (ASAN + COV), which should each be copied into a separate image.

We could potentially shorten by keeping the caches in the same Dockerimage -- however, I don't think we should do this since the next steps is to have caches prepared for replayable builds which need separate images.

Reply:
Yes, these steps sgtm. We'll shortly also adjust this to:

```
# High-level steps:
# 1) Build image for project
# 2) Run an ASAN build and store ccache and generate/validate replay script 
# 2.1) If replay script works, just remove the ccache and commit the container.
# 2.2) If not, Copy ccache cache to host and copy into project's OSS-Fuzz folder
# 2.2.1) Build image for project and copy ccache in, storing image as *-ofg-cache-address
... repeat for coverage 
```

right?

Reply:
Correct